pkgs/development/python-modules/cynthion/default.nix

```diff
@@ -0,0 +1,95 @@
+{
+  lib,
+  fetchFromGitHub,
+  buildPythonPackage,
+  amaranth,
+  pythonOlder,
+  setuptools,
+  future,
+  libusb1,
+  prompt-toolkit,
+  pyfwup,
+  pyserial,
+  pyusb,
+  tabulate,
+  tomli,
+  tqdm,
+  pygreat,
+  apollo-fpga,
+  luna-usb,
+  luna-soc,
+  pytestCheckHook,
+}:
+
+let
+  amaranth_0_4_1 = amaranth.overridePythonAttrs rec {
+    version = "0.4.1";
+
+    src = fetchFromGitHub {
+      owner = "amaranth-lang";
+      repo = "amaranth";
+      rev = "refs/tags/v${version}";
+      sha256 = "sha256-VMgycvxkphdpWIib7aZwh588En145RgYlG2Zfi6nnDo=";
+    };
+
+    postPatch = null;
+  };
```

Comment:
I'm unsure if this would be the correct way of including an older version of `amaranth`.

This particular version is [required](https://github.com/greatscottgadgets/cynthion/blob/0.1.7/cynthion/python/pyproject.toml#L45) and there's a [WIP issue to update amaranth in Cynthion](https://github.com/greatscottgadgets/cynthion/issues/39)

Reply:
We can't override python packages in python-modules due to PYTHONPATH conflicts.
So please mark it as broken, such as `broken = versionAtLeast amaranth.version "0.5"`.
https://github.com/NixOS/nixpkgs/blob/76612b17c0ce71689921ca12d9ffdc9c23ce40b2/pkgs/development/python-modules/tree-sitter-languages/default.nix#L62

Then we could override the python package set in `pkgs/by-name/cynthion/package.nix`, like this.
https://github.com/NixOS/nixpkgs/blob/76612b17c0ce71689921ca12d9ffdc9c23ce40b2/pkgs/by-name/au/autotools-language-server/package.nix#L8-L16